develop/CICD/Skyline Communications/Quality Gate/CICD_Quality_Gate.md

```diff
@@ -0,0 +1,562 @@
+---
+uid: CICD_Quality_Gate_Github
+keywords: CICD, SonarQube, SonarCloud, Quality Gate, Quality
+---
+
+# CI/CD quality gate GitHub
+
+With a CI/CD pipeline, deployment can be automatically blocked if it does not meet certain quality standards. These standards are configured per organization and are sometimes tweaked for specific departments or outputs. They supplement manual efforts in quality control, such as code reviews and manual testing procedures.
+
+Within Skyline Communications' GitHub organization, we strive to reuse the same quality gate for all our coding artifacts (connectors, Automation scripts, libraries, tooling, etc.).
+
+## Valid compilation
+
+If you provide a Visual Studio solution, we will first attempt to compile it on the GitHub runner agents. This often catches assembly mistakes, for example when a developer references code from a third-party assembly but forgets to add that assembly to their repository. If such mistakes were not caught, things might compile okay on the machine of the original developer but would not compile on other machines.
+
+To accomplish this, we run:
+
+```bash
+dotnet build "./NameOfSolution.sln" -p:DefineConstants="DCFv1%3BDBInfo%3BALARM_SQUASHING"
+```
+
+Any compilation failure will trigger the quality gate and block a release.
+
+## Automated tests
+
+We will attempt to run any tests defined next to the source code. These could be written in MSTest, NUnit, or other frameworks. As long as they are recognized by .NET as tests, we will run them.
+
+For GitHub, we currently only run unit tests. For integration tests, we are looking into the concept of DAAS to run such code against.
```

Comment:
Do you mean Daas? If not, what does DAAS stand for here?

Reply:
DataMiner as a Service, yes DaaS